### PR TITLE
Handle url contains # and unsafe parameters

### PR DESCRIPTION
**Pull request by aiceflower**
Handle url contains # and unsafe parameters

### CODE REVIEW

File: linkis-dist/package/conf/linkis.properties [hunk 1]
```diff
@@ -27,6 +27,7 @@ wds.linkis.server.mybatis.datasource.username=
 wds.linkis.server.mybatis.datasource.password=
 # mysql
 wds.linkis.mysql.is.encrypt=false
+linkis.mysql.strong.security.enable=false
```
Comment (casionone) [inline linkis-dist/package/conf/linkis.properties:31]: add info to docs/info-1.3.1.md

File: linkis-engineconn-plugins/jdbc/src/main/java/org/apache/linkis/manager/engineplugin/jdbc/utils/JdbcParamUtils.java [hunk 2]
```diff
@@ -64,20 +71,63 @@ public static void validateJdbcUrl(String url) {
...
+      return url + QUESTION_MARK + APPEND_PARAMS;
+    }
+
+    // enable strong security
+    if (Boolean.valueOf(MYSQL_STRONG_SECURITY_ENABLE.getValue())) {
+      return url + QUESTION_MARK + APPEND_PARAMS;
+    }
+
+    int two = 2;
+
```
Comment (casionone) [inline linkis-engineconn-plugins/jdbc/src/main/java/org/apache/linkis/manager/engineplugin/jdbc/utils/JdbcParamUtils.java:93]: it is not necessary

File: linkis-public-enhancements/linkis-datasource/linkis-metadata-query/service/jdbc/src/main/java/org/apache/linkis/metadata/query/service/mysql/SqlConnection.java [hunk 3]
```diff
@@ -99,6 +116,17 @@ private void validateParams(Map<String, Object> extraParams) {
     extraParams.put("allowUrlInLocalInfile", "false");
   }
 
+  private boolean keyAndValueIsNotSecurity(String key, String value, String param) {
+    return !(isSecurity(key, param) && isSecurity(value, param));
```
Comment (casionone) [inline linkis-public-enhancements/linkis-datasource/linkis-metadata-query/service/jdbc/src/main/java/org/apache/linkis/metadata/query/service/mysql/SqlConnection.java:120]: add some log

File: linkis-engineconn-plugins/jdbc/src/main/java/org/apache/linkis/manager/engineplugin/jdbc/utils/JdbcParamUtils.java [hunk 4]
```diff
@@ -64,20 +71,63 @@ public static void validateJdbcUrl(String url) {
...
-        tmp = tmp + AND_SYMBOL;
+
+    // no params
+    if (!url.contains(String.valueOf(QUESTION_MARK))) {
+      return url + QUESTION_MARK + APPEND_PARAMS;
+    }
+
+    // enable strong security
+    if (Boolean.valueOf(MYSQL_STRONG_SECURITY_ENABLE.getValue())) {
+      return url + QUESTION_MARK + APPEND_PARAMS;
```
Comment (casionone) [inline linkis-engineconn-plugins/jdbc/src/main/java/org/apache/linkis/manager/engineplugin/jdbc/utils/JdbcParamUtils.java:89]: add some log

**Review by CCweixiao — APPROVED** [review]

**Pull request merged**